### PR TITLE
Add AuxKernel to output volumetric heating from Joule heating 

### LINKED ISSUE #12723: Add an AuxKernel to output volumetric heating due to Joule heating
## Rationale

We need to output the spatial distribution and the integrated value of the heat generated by the JouleHeatingSource kernel in the heat_conduction module. 
## Description

This can be done by adding an AuxKernel that outputs the value computed by the JouleHeatingSource kernel.

## Impact

No impact on existing code.  This will simply add a new object.

### PR DESCRIPTION
**Pull request by yeh512**
closes #12723

### CODE REVIEW

**Comment by bwspenc** [comment]
@yeh512 There are a lot of extra commits here.  My guess is that you did your work on the master branch rather than the devel branch.  I'll try to swing by and help you sort that out.
Also, you'll need to run 'git clang-format'.

File: modules/heat_conduction/src/JouleHeatingHeatGeneratedAux.C [hunk 1]
```diff
@@ -0,0 +1,41 @@
+//* This file is part of the MOOSE framework
```
Comment (permcody) [inline modules/heat_conduction/src/JouleHeatingHeatGeneratedAux.C]: Please create an "auxkernels" directory under source and move this file into that folder.

File: modules/heat_conduction/include/kernels/JouleHeatingHeatGeneratedAux.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+//* This file is part of the MOOSE framework
```
Comment (permcody) [inline modules/heat_conduction/include/kernels/JouleHeatingHeatGeneratedAux.h]: Create an "auxkernels" folder under include and move this file into that folder.

File: modules/heat_conduction/doc/content/source/kernels/JouleHeatingHeatGeneratedAux.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#JouleHeatingHeatGeneratedAux
```
Comment (permcody) [inline modules/heat_conduction/doc/content/source/kernels/JouleHeatingHeatGeneratedAux.md]: Create an "auxkernels" folder under source and move this file into that location.

File: modules/combined/test/tests/joule_heating_heat/tests [hunk 4]
```diff
@@ -0,0 +1,12 @@
+[Tests]
+  issues = '12723'
```
Comment (permcody) [inline modules/combined/test/tests/joule_heating_heat/tests]: 12723 -> #12723

File: modules/combined/test/tests/joule_heating_heat/tests [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+  design = 'kernels/JouleHeatingHeatGeneratedAux.md'
```
Comment (permcody) [inline modules/combined/test/tests/joule_heating_heat/tests]: Update this once you move this into the new auxkernels folder.

File: modules/combined/test/tests/joule_heating_heat/jouleheating_heat.i [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  nl_rel_tol = 1e-4
+  petsc_options_iname = '-pc_type -pc_hypre_type'
+  petsc_options_value = 'hypre boomeramg'
+  dt = 0.5
+  end_time = 5
+[]
+
+[Outputs]
+  exodus = true
+  execute_on = 'initial timestep_end'
```
Comment (permcody) [inline modules/combined/test/tests/joule_heating_heat/jouleheating_heat.i]: This is the default, you can remove this line if you want.

**Review by permcody — CHANGES_REQUESTED** [review]
Looks really good, just a few minor changes and hopefully the Documentation target will pass!

**Comment by bwspenc** [comment]
The .h file should be in:
```
modules/heat_conduction/include/auxkernels/
```
and the .C file should be in:
```
modules/heat_conduction/src/auxkernels/

**Review by permcody — APPROVED** [review]
Very nice - Thanks for the contribution!

**Pull request merged**